### ISSUE TITLE
RTX3090 performs no better than 1080ti

### ISSUE THREAD

**Issue by antsfamily** [pytorch/pytorch#50328]
# Questions

I want to compare the performance of Pytorch on 3090 and 1080ti graphics cards. I find that 3090 is no better than 1080ti when training a network and 1x faster than 1080ti when testing a network. Why?

# Hardware and software configuration

- GPU: ``RTX3090`` (MSI VENTUS 3X 24G OC), ``GTX1080ti``
- CPU: ``E5-2696v3``
- Memory: ``64G``
- PCI version: 3.0
- Power Supply: 1000W
- CUDA: ``11.1``
- CUDNN: ``8005``
- Pytorch: ``1.8.0.dev20210106+cu110``

# Results

## MNIST classification with a CNN

```bash
Torch Version:  1.8.0.dev20210106+cu110
Torch CUDA Version:  11.0
CUDNN Version:  8005
GPU Device:  GeForce RTX 3090
CUDNN Benchmark:  False
CUDNN Deterministic:  True
CUDA TF32:  True
CUDNN TF32:  True
Workers:  4
Batch Size:  64
Epochs:  10
Training Loss:  0.0009187350056997093
Testing  Loss:  0.03026894662413977
Time: 63.0998



Torch Version:  1.8.0.dev20210106+cu110
Torch CUDA Version:  11.0
CUDNN Version:  8005
GPU Device:  GeForce GTX 1080 Ti
CUDNN Benchmark:  False
CUDNN Deterministic:  True
CUDA TF32:  True
CUDNN TF32:  True
Workers:  4
Batch Size:  64
Epochs:  10
Training Loss:  0.0008879667648342487
Testing  Loss:  0.030606915746741615
Time: 56.9057
```

## Convolution

```bash
Torch version:  1.8.0.dev20210106+cu110
Torch CUDA version:  11.0
CUDNN Version:  8005
GeForce RTX 3090
Conv1d model, benchmark=False, deterministic=False, 0.687ms/iter
Conv1d model, benchmark=True, deterministic=False, 0.511ms/iter
Conv1d model, benchmark=False, deterministic=True, 0.540ms/iter
Conv1d model, benchmark=True, deterministic=True, 0.484ms/iter
Conv2d model, benchmark=False, deterministic=False, 1.327ms/iter
Conv2d model, benchmark=True, deterministic=False, 1.335ms/iter
Conv2d model, benchmark=False, deterministic=True, 1.474ms/iter
Conv2d model, benchmark=True, deterministic=True, 1.480ms/iter
Conv2d1 model, benchmark=False, deterministic=False, 3.278ms/iter
Conv2d1 model, benchmark=True, deterministic=False, 3.280ms/iter
Conv2d1 model, benchmark=False, deterministic=True, 3.286ms/iter
Conv2d1 model, benchmark=True, deterministic=True, 3.286ms/iter

Torch version:  1.8.0.dev20210106+cu110
Torch CUDA version:  11.0
CUDNN Version:  8005
GeForce GTX 1080 Ti
Conv1d model, benchmark=False, deterministic=False, 0.709ms/iter
Conv1d model, benchmark=True, deterministic=False, 0.711ms/iter
Conv1d model, benchmark=False, deterministic=True, 0.844ms/iter
Conv1d model, benchmark=True, deterministic=True, 0.711ms/iter
Conv2d model, benchmark=False, deterministic=False, 2.684ms/iter
Conv2d model, benchmark=True, deterministic=False, 2.883ms/iter
Conv2d model, benchmark=False, deterministic=True, 2.212ms/iter
Conv2d model, benchmark=True, deterministic=True, 2.195ms/iter
Conv2d1 model, benchmark=False, deterministic=False, 5.583ms/iter
Conv2d1 model, benchmark=True, deterministic=False, 6.077ms/iter
Conv2d1 model, benchmark=False, deterministic=True, 6.097ms/iter
Conv2d1 model, benchmark=True, deterministic=True, 6.120ms/iter
```

# Code
 
## MNIST classification with a CNN

The code can be downloaded from [here](https://github.com/antsfamily/torchlib/blob/master/tests/device/th_mnist_cnn_device.py).

## Convolution

The code can be downloaded from [here](https://github.com/antsfamily/torchlib/blob/master/tests/device/th_conv_device.py).

**Comment by mahirkukreja** [comment]
Performance of Pytorch on RTX 3080 -

Torch version:  1.7.1+cu110
Torch CUDA version:  11.0
CUDNN Version:  8005
GeForce RTX 3080

MNIST classification with a CNN

Training Loss:  0.0009187350056997093
Testing  Loss:  0.03026894662413977
Time: 30.7122

Convolution

Conv1d model, benchmark=False, deterministic=False, 0.271ms/iter
Conv1d model, benchmark=True, deterministic=False, 0.271ms/iter
Conv1d model, benchmark=False, deterministic=True, 0.298ms/iter
Conv1d model, benchmark=True, deterministic=True, 0.274ms/iter
Conv2d model, benchmark=False, deterministic=False, 1.487ms/iter
Conv2d model, benchmark=True, deterministic=False, 1.490ms/iter
Conv2d model, benchmark=False, deterministic=True, 1.680ms/iter
Conv2d model, benchmark=True, deterministic=True, 1.678ms/iter
Conv2d1 model, benchmark=False, deterministic=False, 3.757ms/iter
Conv2d1 model, benchmark=True, deterministic=False, 3.763ms/iter
Conv2d1 model, benchmark=False, deterministic=True, 3.756ms/iter
Conv2d1 model, benchmark=True, deterministic=True, 3.755ms/iter

**Comment by ngimel** [comment]
cc @xwang233. This is a very small model dominated by overheads, you won't see a lot of perf benefits from 3090 for it.

**Comment by antsfamily** [comment]
@mahirkukreja，thanks very much，but I can not see your configuration for  #"Workers", "Batch Size", 
and "Epochs". What's more, what is the version of PCIE on your motherboard.

**Comment by mahirkukreja** [comment]
> @mahirkukreja，thanks very much，but I can not see your configuration for #"Workers", "Batch Size",
> and "Epochs". What's more, what is the version of PCIE on your motherboard.

PCIE Version is 3.0.
I used the default parameters defined in your script without making any changes.

**Comment by VvvvvGH** [comment]
9900K + 3090 on Windows
Cuda 11.2 build
```
Torch version:  1.7.0a0+57bffc3
Torch CUDA version:  11.2
CUDNN Version:  8005
GeForce RTX 3090
Conv1d model, benchmark=False, deterministic=False, 0.307ms/iter
Conv1d model, benchmark=True, deterministic=False, 0.307ms/iter
Conv1d model, benchmark=False, deterministic=True, 0.363ms/iter
Conv1d model, benchmark=True, deterministic=True, 0.307ms/iter
Conv2d model, benchmark=False, deterministic=False, 1.387ms/iter
Conv2d model, benchmark=True, deterministic=False, 1.388ms/iter
Conv2d model, benchmark=False, deterministic=True, 1.481ms/iter
Conv2d model, benchmark=True, deterministic=True, 1.482ms/iter
Conv2d1 model, benchmark=False, deterministic=False, 3.847ms/iter
Conv2d1 model, benchmark=True, deterministic=False, 3.848ms/iter
Conv2d1 model, benchmark=False, deterministic=True, 3.849ms/iter
Conv2d1 model, benchmark=True, deterministic=True, 4.203ms/iter
```

**Comment by wellhowtosay** [comment]
> 9900K + 3090 on Windows
> Cuda 11.2 build
> 
> ```
> Torch version:  1.7.0a0+57bffc3
> Torch CUDA version:  11.2
> CUDNN Version:  8005
> GeForce RTX 3090
> Conv1d model, benchmark=False, deterministic=False, 0.307ms/iter
> Conv1d model, benchmark=True, deterministic=False, 0.307ms/iter
> Conv1d model, benchmark=False, deterministic=True, 0.363ms/iter
> Conv1d model, benchmark=True, deterministic=True, 0.307ms/iter
> Conv2d model, benchmark=False, deterministic=False, 1.387ms/iter
> Conv2d model, benchmark=True, deterministic=False, 1.388ms/iter
> Conv2d model, benchmark=False, deterministic=True, 1.481ms/iter
> Conv2d model, benchmark=True, deterministic=True, 1.482ms/iter
> Conv2d1 model, benchmark=False, deterministic=False, 3.847ms/iter
> Conv2d1 model, benchmark=True, deterministic=False, 3.848ms/iter
> Conv2d1 model, benchmark=False, deterministic=True, 3.849ms/iter
> Conv2d1 model, benchmark=True, deterministic=True, 4.203ms/iter
> ```

how did you install pytorch, build from source?

**Comment by dimitree54** [comment]
I am also facing this issue:
On pip installed pytorch 1.4 + RTX 2080:
```
Torch Version:  1.4.0
Torch CUDA Version:  10.1
CUDNN Version:  7603
GPU Device:  GeForce RTX 2080
CUDNN Benchmark:  False
CUDNN Deterministic:  True
CUDA TF32:  True
CUDNN TF32:  True
Workers:  4
Batch Size:  64
Epochs:  10
Training Loss:  0.0009221168732586859
Testing  Loss:  0.0317381218124926
Time: 46.9702
```
And GPU utilization is ~80% during script run

On compiled pytorch 1.8 + RTX 3090:
```
Torch Version:  1.8.0a0+e34992e
Torch CUDA Version:  11.1
CUDNN Version:  8005
GPU Device:  GeForce RTX 3090
CUDNN Benchmark:  False
CUDNN Deterministic:  True
CUDA TF32:  True
CUDNN TF32:  True
Workers:  4
Batch Size:  64
Epochs:  10
Training Loss:  0.0008818266955849929
Testing  Loss:  0.028839084201603866
Time: 70.8309
```

**Comment by antsfamily** [comment]
The performance of 3090 on training a large network is worse than 1080ti. RTX 3090 takes twice as much time as 1080ti.

```bash
# Results on 3090
--->Train epoch: 41, loss: -4.4800, entropy: 9.8360, time: 298.0467309951782s
--->Valid epoch: 41, loss: -2.3898, entropy: 10.2245, time: 31.563281536102295s
--->Test epoch: 41, loss: -4.6982, entropy: 9.8963, time: 31.70620560646057s
--->Train epoch: 42, loss: -4.5034, entropy: 9.8335, time: 297.7266011238098s
--->Valid epoch: 42, loss: -2.3596, entropy: 10.2282, time: 31.558250665664673s
--->Test epoch: 42, loss: -4.5927, entropy: 9.9066, time: 31.6065571308136s

# Results on 1080ti
--->Train epoch: 41, loss: -4.4862, entropy: 9.8349, time: 126.39260697364807s
--->Valid epoch: 41, loss: -2.4010, entropy: 10.2218, time: 40.516133069992065s
--->Test epoch: 41, loss: -4.7926, entropy: 9.8841, time: 40.41139888763428s
--->Train epoch: 42, loss: -4.4698, entropy: 9.8365, time: 126.09351229667664s
--->Valid epoch: 42, loss: -2.3446, entropy: 10.2333, time: 39.89955806732178s
--->Test epoch: 42, loss: -4.5896, entropy: 9.9056, time: 40.008435010910034s
```

**Comment by slaweks17** [comment]
I experience the same: on Pytorch 1.9 or 1.8, 1080ti is between 20 to 80% faster than 3090. My model comprises a few rather large (state=250) LSTMs.

**Comment by antsfamily** [comment]
> I experience the same: on Pytorch 1.9 or 1.8, 1080ti is between 20 to 80% faster than 3090. My model comprises a few rather large (state=250) LSTMs.

When training CNNs, RTX3090 performs normally with Pytorch version > 1.8.0 (such as 1.8.1 or 1.9).

**Comment by slaweks17** [comment]
I am using Pytorch 1.9(stable,  CUDA 11.1, on Windows 10.

**Comment by FrancescoSaverioZuppichini** [comment]
same here
